File: Square_SortedArr/square.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public int[] SortedSquares(int[] nums)
    {
        int[] result = new int[nums.Length];

        int i = 0,
            j = nums.Length - 1,
            index = nums.Length - 1;

        while (i <= j)
        {
            int iSquare = nums[i] * nums[i];
            int jSquare = nums[j] * nums[j];
            if (iSquare > jSquare)
            {
                result[index] = iSquare;
                i++;
            }
            else
            {
                result[index] = jSquare;
                j--;
            }

            index--;
        }

        return result;
    }
}

File: Square_SortedArr/square.go
package squaresortedarr

func sortedSquares(nums []int) []int {
	result := make([]int, len(nums))

	i, j, index := 0, len(nums)-1, len(nums)-1

	for i <= j {
		iSquare := nums[i] * nums[i]
		jSquare := nums[j] * nums[j]

		if iSquare > jSquare {
			result[index] = iSquare
			i++
		} else {
			result[index] = jSquare
			j--
		}
		index--
	}

	return result
}
